mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp

```diff
@@ -46,6 +46,10 @@ void miopen::addHighLevelPipeline(PassManager &pm, bool toMIOpen) {
     pm.addPass(tosa::createTosaToMIOpenPass());
   }
   mlir::tosa::addTosaToLinalgPasses(pm);
+  bufferization::OneShotBufferizationOptions options;
+  options.allowReturnAllocs = true;
+  options.testAnalysisOnly = true;
```

Comment:
Why is testAnalysisOnly necessary. From the description it seems to suggest that it won't do any of the transforms.

Reply:
By looking at their pass pipeline:

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/579526e08448ce42d89528f75c870175c337c34b/external/llvm-project/mlir/lib/Dialect/Linalg/Transforms/ComprehensiveBufferizePass.cpp#L117-L124

Everything below `opt.testAnalysisOnly` will break MIOpen dialect and the reason is not yet clear to me. But I can confirm that everything before the escape is equivalent to previous pipeline.